feat: Add CloudQuery integration (ECOINT-11)

erezrokah opened the pull request:
### What does this PR do?

Add [CloudQuery](https://www.cloudquery.io/) integration

### Motivation

Get listed on the integrations page per https://docs.datadoghq.com/developers/integrations/?tab=integrations

### Review checklist

- [x] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [x] Feature or bugfix has tests
- [x] Git history is clean
- [x] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)
- [x] If this PR includes a log pipeline, please add a description describing the remappers and processors. 

### Additional Notes

Anything else we should know when reviewing?

I followed the guide in https://docs.datadoghq.com/developers/integrations/?tab=integrations as best as I can. Feedback welcomed!

drichards-87 commented:
Created Jira card for Docs Team review.

estherk15 commented on cloudquery/README.md:
```diff
@@ -0,0 +1,177 @@
+
+> For additional ways to run the collector, please refer to the [official documentation][7].
+
+- Direct OTEL Ingestion by the Datadog Agent via a configuration file
```
```suggestion
#### Datadog Agent OTEL ingestion through a configuration file
```

estherk15 commented on cloudquery/README.md:
```diff
@@ -0,0 +1,177 @@
+
+- Direct OTEL Ingestion by the Datadog Agent via a configuration file
+
+[Locate][8] your `datadog.yaml` file and add the following configuration:
```
```suggestion
1. Locate your [`datadog.yaml` Agent configuration file][8] and add the following configuration:
```

estherk15 commented on cloudquery/README.md:
````diff
@@ -0,0 +1,177 @@
+logs_enabled: true
+```
+
+[Restart][9] the Datadog agent for the change to take effect.
````
```suggestion
1. [Restart][9] the Datadog agent for the change to take effect.
```

estherk15 commented on cloudquery/README.md:
```diff
@@ -0,0 +1,177 @@
+
+[Restart][9] the Datadog agent for the change to take effect.
+
+- Direct OTEL ingestion by the Datadog Agent via environment variables
```
```suggestion
#### Datadog Agent OTEL ingestion through environment variables
```

estherk15 commented on cloudquery/README.md:
```diff
@@ -0,0 +1,177 @@
+
+- Direct OTEL ingestion by the Datadog Agent via environment variables
+
+Pass the `DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_HTTP_ENDPOINT` environment variable to the Datadog agent with a value of `0.0.0.0:4318`.
```
```suggestion
1. Pass the `DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_HTTP_ENDPOINT` environment variable to the Datadog Agent with a value of `0.0.0.0:4318`.
```

estherk15 commented on cloudquery/README.md:
```diff
@@ -0,0 +1,177 @@
+- Direct OTEL ingestion by the Datadog Agent via environment variables
+
+Pass the `DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_HTTP_ENDPOINT` environment variable to the Datadog agent with a value of `0.0.0.0:4318`.
+If you're using Docker compose, you can find an example below:
```
```suggestion
If you're using Docker compose, see the example below:
```

estherk15 commented on cloudquery/README.md:
```diff
@@ -0,0 +1,177 @@
+[7]: https://docs.datadoghq.com/opentelemetry/collector_exporter/deployment#running-the-collector
+[8]: https://docs.datadoghq.com/agent/configuration/agent-configuration-files/
+[9]: https://docs.datadoghq.com/agent/configuration/agent-commands/#restart-the-agent
+[10]: https://docs.datadoghq.com/agent/configuration/agent-commands/#restart-the-agent
```
duplicate

```suggestion
```

estherk15 commented on cloudquery/README.md:
````diff
@@ -0,0 +1,177 @@
+      - "4318:4318"
+```
+
+[Restart][10] the Datadog agent for the change to take effect.
````
```suggestion
1. [Restart][9] the Datadog agent for the change to take effect.
```

estherk15 commented on cloudquery/README.md:
```diff
@@ -0,0 +1,177 @@
+
+[Restart][10] the Datadog agent for the change to take effect.
+
+> For additional ways to configure the Datadog agent, please refer to the [official documentation][11].
```
```suggestion
For more ways to configure the Datadog Agent, see [OTLP Ingestion by the Datadog Agent][11].
```

estherk15 commented on cloudquery/README.md:
```diff
@@ -0,0 +1,177 @@
+
+> For additional ways to configure the Datadog agent, please refer to the [official documentation][11].
+
+Once you have the agent or collector ready, you can specify the endpoint in the source spec:
```
Is this optional? If not, recommend moving this to each configuration as the final step. For example: 
```
1. Restart the Datadog agent for the change to take effect.
2. After you have the Agent ready, specify the endpoint in the source spec:
```
```suggestion
After you have the Agent or collector ready, specify the endpoint in the source spec:
```

erezrokah replied:
This is required, good suggestion 👍

estherk15 commented on cloudquery/README.md:
```diff
@@ -0,0 +1,177 @@
+### Validation
+
+Run `cloudquery sync spec.yml`.
+Once ingestion starts you should be able to start seeing the traces in Datadog under APM->Traces->Explorer.
```
```suggestion
After ingestion starts, you should start seeing the traces in the Datadog [**APM Traces Explorer**][13].
```

estherk15 commented on cloudquery/README.md:
```diff
@@ -0,0 +1,177 @@
+
+Run `cloudquery sync spec.yml`.
+Once ingestion starts you should be able to start seeing the traces in Datadog under APM->Traces->Explorer.
+You should see metrics and logs in the respective sections of the Datadog UI as well.
```
```suggestion
You can also validate metrics and logs in the [**Metrics Summary**][14] and [**Log Explorer**][15].
```

estherk15 commented on cloudquery/README.md:
```diff
@@ -0,0 +1,177 @@
+## Uninstallation
+
+If you use the OpenTelemetry collector, you can stop it by running `docker stop <container_id>`.
+IF you use the Datadog agent, remove the configuration or environment variables you added and [restart][10] the agent.
```
```suggestion
If you use the Datadog Agent, remove the configuration or environment variables you added and [restart][9] the agent.
```

estherk15 commented on cloudquery/README.md:
```diff
@@ -0,0 +1,177 @@
+
+## Support
+
+Need help? [Contact us][4].
```
```suggestion
For support, contact [CloudQuery][4].
```

estherk15 commented on cloudquery/assets/dashboards/cloudquery_syncs_insights.json:
```diff
@@ -0,0 +1,324 @@
+            "id": 1803149444713820,
+            "definition": {
+              "type": "note",
+              "content": "Use this dashboard to get insights into CloudQuery syncs and optimize them. For more information on setting up CloudQuery with Datadog visit our [docs](https://docs.cloudquery.io/docs/advanced-topics/monitoring#opentelemetry-and-datadog).",
```
```suggestion
              "content": "Use this dashboard to get insights into CloudQuery syncs and optimize them. For more information on setting up CloudQuery with Datadog, see the [CloudQuery documentation](https://docs.cloudquery.io/docs/advanced-topics/monitoring#opentelemetry-and-datadog).",
```

estherk15 reviewed (commented):
Left some suggestions for documentation, let me know if you have any questions or would like a re-review!

erezrokah commented on cloudquery/README.md:
```diff
@@ -0,0 +1,177 @@
+### Configuration
+
+CloudQuery supports [OpenTelemetry][5] traces, metrics and logs out of the box.
+There are multiple ways to configure OpenTelemetry with Datadog. We'll show only a subset of them here, and you can find more information [in the Datadog docs][6].
```
Great suggestion, wasn't sure if I can use more nesting for the headers (e.g. `####`)

erezrokah commented:
Thanks for the review @estherk15, great suggestions 🚀 I also followed up with https://github.com/DataDog/integrations-extras/pull/2451/commits/087f84ae9d7591c359601eb8629ba28d536e5c65 and https://github.com/DataDog/integrations-extras/pull/2451/commits/06a21eef487b51742391b4bad2a1911624719eb5 to fix the links and lists numbering (as the lists numbering is not preserved after the code blocks).

estherk15 commented on cloudquery/README.md:
````diff
@@ -0,0 +1,188 @@
+    -v $(pwd)/otel_collector_config.yaml:/etc/otelcol-contrib/config.yaml \
+    otel/opentelemetry-collector-contrib:0.104.0
+```
+
````
````suggestion
3. After you have the collector ready, specify the endpoint in the source spec:

```yaml
kind: source
spec:
  name: "aws"
  path: "cloudquery/aws"
  registry: "cloudquery"
  version: "VERSION_SOURCE_AWS"
  tables: ["aws_s3_buckets"]
  destinations: ["postgresql"]
  otel_endpoint: "0.0.0.0:4318"
  otel_endpoint_insecure: true
  spec:
```

````

estherk15 commented on cloudquery/README.md:
````diff
@@ -0,0 +1,188 @@
+logs_enabled: true
+```
+
+2. [Restart][9] the Datadog agent for the change to take effect.
````
````suggestion
2. [Restart][9] the Datadog agent for the change to take effect.

3. After you have the Agent ready, specify the endpoint in the source spec:

```yaml
kind: source
spec:
  name: "aws"
  path: "cloudquery/aws"
  registry: "cloudquery"
  version: "VERSION_SOURCE_AWS"
  tables: ["aws_s3_buckets"]
  destinations: ["postgresql"]
  otel_endpoint: "0.0.0.0:4318"
  otel_endpoint_insecure: true
  spec:
```
````

estherk15 commented on cloudquery/README.md:
````diff
@@ -0,0 +1,188 @@
+      - "4318:4318"
+```
+
+2. [Restart][9] the Datadog agent for the change to take effect.
````
````suggestion
2. [Restart][9] the Datadog agent for the change to take effect.

3. After you have the Agent ready, specify the endpoint in the source spec:
```yaml
kind: source
spec:
  name: "aws"
  path: "cloudquery/aws"
  registry: "cloudquery"
  version: "VERSION_SOURCE_AWS"
  tables: ["aws_s3_buckets"]
  destinations: ["postgresql"]
  otel_endpoint: "0.0.0.0:4318"
  otel_endpoint_insecure: true
  spec:
```
````

estherk15 commented on cloudquery/README.md:
````diff
@@ -0,0 +1,188 @@
+
+For more ways to configure the Datadog Agent, see [OTLP Ingestion by the Datadog Agent][10].
+
+After you have the Agent or collector ready, specify the endpoint in the source spec:
+
+```yaml
+kind: source
+spec:
+  name: "aws"
+  path: "cloudquery/aws"
+  registry: "cloudquery"
+  version: "VERSION_SOURCE_AWS"
+  tables: ["aws_s3_buckets"]
+  destinations: ["postgresql"]
+  otel_endpoint: "0.0.0.0:4318"
+  otel_endpoint_insecure: true
+  spec:
+```
````
```suggestion
```

estherk15 reviewed (approved):
Thanks for taking my suggestions! Looks great!

erezrokah commented:
> Thanks for taking my suggestions! Looks great!

Thanks for reviewing, I added another commit in https://github.com/DataDog/integrations-extras/pull/2451/commits/ac03e1c416792a36896ec93b7dd3b971cc88cc47 to better explain the version placeholders. I also removed the `registry: cloudquery` as that's the default and optional in our configuration

erezrokah commented:
Thanks @estherk15, per a meeting with the team in charge of partner integrations, we'll submit this as a community integration until there's official OTEL support

erezrokah commented:
> we'll submit this as a community integration until there's official OTEL support

OK so I guess I got confused by something, since the instructions in https://docs.datadoghq.com/agent/guide/use-community-integrations/?tab=agentv721v621 say to open a PR to this repo.

Can we accept this PR as a community integration?

erezrokah commented:
Moving to draft until I fix the log pipelines CI

erezrokah commented:
Not sure how to fix these errors:
![image](https://github.com/user-attachments/assets/50fd82df-1e0d-4120-9c1c-a84057964798)

This is how I configured the filter in the Datadog UI for AWS:
![image](https://github.com/user-attachments/assets/01bd3db7-db4b-4bce-b94a-46cef9732ce7)

We send a `service` field with `name` and `version` properties in the logs so we should be able to filter by those no?

erezrokah commented:
OK, removed the log pipelines in the initial version

erezrokah commented:
https://github.com/DataDog/integrations-extras/pull/2451/commits/14209ecb539754e70796c5651d4f271b4f85e072 marks this as a community integration

HadhemiDD commented:
Can you please fix the conflict in CODEOWNERS file? you just need to rebase it with master as it has conflicts with it.

erezrokah commented:
> Can you please fix the conflict in CODEOWNERS file? you just need to rebase it with master as it has conflicts with it.

Done! Thanks for the bump
